components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -808,7 +826,234 @@ public ConnectedApps getConnectedApps(String resourceId, Integer limit, Integer
+            return createIDPTemplateResponse(idpTemplate);
+        } catch (TemplateManagementException e) {
+            throw handleTemplateMgtException(e, Constants.ErrorMessage.ERROR_CODE_ERROR_RETRIEVING_IDP_TEMPLATE,
+                    templateId);
+        } catch (IOException e) {
+            throw handleException(Response.Status.INTERNAL_SERVER_ERROR,
+                    Constants.ErrorMessage.ERROR_CODE_ERROR_RETRIEVING_IDP_TEMPLATE, templateId);
+        }
+    }
+
+    /**
+     * Update IDP template.
+     *
+     * @param identityProviderTemplate Updated IDP template
+     */
+    public void updateIDPTemplate(String templateId, IdentityProviderTemplate identityProviderTemplate) {
+
+        try {
+            Template idpTemplate = generateIDPTemplate(identityProviderTemplate);
+            IdentityProviderServiceHolder.getTemplateManager().updateTemplateById(templateId, idpTemplate);
+        } catch (TemplateManagementException e) {
+            throw handleTemplateMgtException(e, Constants.ErrorMessage.ERROR_CODE_ERROR_UPDATING_IDP_TEMPLATE,
+                    identityProviderTemplate.getId());
+        } catch (JsonProcessingException e) {
+            throw handleException(Response.Status.BAD_REQUEST,
+                    Constants.ErrorMessage.ERROR_CODE_ERROR_UPDATING_IDP_TEMPLATE, templateId);
+        }
+    }
+
+    /**
+     * Create a new IDP template.
+     *
+     * @param identityProviderTemplate identityProviderTemplatePOSTRequest
+     * @return IdentityProviderTemplateResponse
+     */
+    public String createIDPTemplate(IdentityProviderTemplate identityProviderTemplate) {
+
+        try {
+            TemplateManager templateManager = IdentityProviderServiceHolder.getTemplateManager();
+            Template idpTemplate = generateIDPTemplate(identityProviderTemplate);
+            return templateManager.addTemplate(idpTemplate);
+        } catch (TemplateManagementException e) {
+            throw handleTemplateMgtException(e, Constants.ErrorMessage.ERROR_CODE_ERROR_ADDING_IDP_TEMPLATE, null);
+        } catch (JsonProcessingException e) {
+            throw handleException(Response.Status.BAD_REQUEST,
+                    Constants.ErrorMessage.ERROR_CODE_ERROR_ADDING_IDP_TEMPLATE, null);
+        }
+    }
+
+    /**
+     * Delete a IDP template identified by resource Id.
+     *
+     * @param templateId Id of the IDP template
+     */
+    public void deleteIDPTemplate(String templateId) {
+
+        try {
+            TemplateManager templateManager = IdentityProviderServiceHolder.getTemplateManager();
+            templateManager.deleteTemplateById(templateId);
+        } catch (TemplateManagementException e) {
+            throw handleTemplateMgtException(e, Constants.ErrorMessage.ERROR_CODE_ERROR_DELETING_IDP_TEMPLATE,
+                    templateId);
+        }
+    }
+
+//  Private utility Methods.
```

Comment:
This type of comment is not the normal convention. Ideally each method needs to have the comment explaining it is a utility and what it is for.

Reply:
Fixed with https://github.com/wso2/identity-api-server/pull/191